perf-tool/src/monitor.cpp

```diff
@@ -138,6 +93,57 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
             j["stackTrace"] = jMethods;
         }
     }
+    return;
+}
+
+JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env, jthread thread, jobject object){
+    json j;
+    jvmtiError error;
+
+    /* Get number of events and increment */
+    int numSamples = atomic_fetch_add(&monitorSampleCount, 1);
+    if (numSamples % monitorConfig.getSampleRate() != 0)
+        return;
+
+    jvmtiError err;
+    jint hash;
+    err = jvmtiEnv->GetObjectHashCode(object, &hash);
+    if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve object hashcode.")) {
+	    char str[32];
+	    sprintf(str, "0x%x", hash);
+	    j["monitorHash"] = str;
+    }
+
+    static std::map<std::string, int> numContentions;
+    jclass cls = env->GetObjectClass(object);
+    /* First get the class object */
+    jmethodID mid = env->GetMethodID(cls, "getClass", "()Ljava/lang/Class;");
+    jobject clsObj = env->CallObjectMethod(object, mid);
```

Comment:
Every `CallObjectMethod` call needs to be followed by an `ExceptionCheck` or clearing of the potential exception.

It's a good idea to use jnicheck to validate the JNI code being written